cime_config/runseq/runseq_general.py

```diff
@@ -59,18 +46,32 @@ def gen_runseq(case, coupling_times):
     if rof_cpl_time < ocn_cpl_time:
         expect(False, "assume that rof_cpl_time is always greater than or equal to ocn_cpl_time")
 
+    if run_glc:
+        # It wouldn't make sense to run GLC unless we also do MED -> GLC to transfer fields to GLC,
+        # and some of the below logic controlling what appears in the run sequence depends on this
+        # (i.e., depends on the fact that, if run_glc is True, then med_to_glc is also True).
+        expect(med_to_glc, "if run_glc is True, then med_to_glc must also be True")
+
     rof_outer_loop = run_rof and rof_cpl_time > atm_cpl_time
     ocn_outer_loop = run_ocn and ocn_cpl_time > atm_cpl_time
 
+    # Note that we do some aspects of the GLC outer loop even if run_glc is False
+    # (as long as med_to_glc is True).
+    #
+    # Note that, in contrast to the other outer_loop variables, this doesn't check glc_cpl_time.
+    # This is for consistency with the logic that was in place before adding this variable;
+    # this seems to implicitly assume that glc_cpl_time > atm_cpl_time.
+    glc_outer_loop = med_to_glc
+
     inner_loop = ((atm_cpl_time < ocn_cpl_time) or
                   (atm_cpl_time < rof_cpl_time) or
-                  (run_glc and atm_cpl_time < glc_cpl_time) or
+                  (glc_outer_loop and atm_cpl_time < glc_cpl_time) or
                   atm_cpl_time == ocn_cpl_time)
 
     with RunSeq(os.path.join(caseroot, "CaseDocs", "nuopc.runseq")) as runseq:
 
         #------------------
-        runseq.enter_time_loop(glc_cpl_time, newtime=run_glc, active=med_to_glc)
+        runseq.enter_time_loop(glc_cpl_time, newtime=glc_outer_loop)
```

Comment:
This piece (the control over the `inner_loop` and `enter_time_loop`'s `newtime`) is the piece I'm least certain of and would be most happy to get a second set of eyes on. These now depend on med_to_glc rather than run_glc – though note that I have effectively swapped the logic for run_glc and med_to_glc in this PR: in most cases they have the same value, but in a NOEVOLVE case now med_to_glc is True and run_glc is False, whereas previously the reverse was the case. It seems right to have these depend on the one that is more often True (med_to_glc now).